ambry-tools/src/test/java/com.github.ambry/store/BlobIdTransformerTest.java

```diff
@@ -0,0 +1,388 @@
+    try {
+      transformer.transform(message);
+      fail("Did not throw NullPointerException");
+    } catch (NullPointerException e) {
+      //expected
+    }
+    //null messageInfo
+    message = new StoreCopier.Message(null, new ByteArrayInputStream(new byte[30]));
+    try {
+      transformer.transform(message);
+      fail("Did not throw NullPointerException");
+    } catch (NullPointerException e) {
+      //expected
+    }
+  }
+
+  private BlobId createBlobId(String hexBlobId) throws IOException {
+    return new BlobId(hexBlobId, clusterMap);
+  }
+
+  private void putRowInStoreKeyConverter(MockStoreKeyConverter storeKeyConverter, BlobIdStringPair blobIdStringPair) throws IOException {
+    storeKeyConverter.put(createBlobId(blobIdStringPair.getUnconverted()), createBlobId(blobIdStringPair.getConverted()));
+  }
+
+  private MockStoreKeyConverter createAndSetupMockStoreKeyConverter(BlobIdStringPair[] blobIdStringPairs) throws IOException {
+    MockStoreKeyConverter storeKeyConverter = new MockStoreKeyConverter();
+    for (BlobIdStringPair blobIdStringPair : blobIdStringPairs) {
+      putRowInStoreKeyConverter(storeKeyConverter, blobIdStringPair);
+    }
+    return storeKeyConverter;
+  }
+
+  private String notEqualPrint(int loc, byte[] actual, byte[] expect) {
+    StringBuilder sb = new StringBuilder();
+    sb.append("Not equal at byte ")
+        .append(loc)
+        .append("\n")
+        .append("Actual: ")
+        .append(Hex.encodeHexString(actual))
+        .append("\n")
+        .append("Expect: ")
+        .append(Hex.encodeHexString(expect))
+        .append("\n");
+    return sb.toString();
+  }
+
+  private void assertInputStreamEqual(InputStream actual, InputStream expect) throws IOException {
+    byte[] actualBuf = inputStreamToByteArray(actual);
+    byte[] expectBuf = inputStreamToByteArray(expect);
+    assertTrue("Lengths are not equal, actual: " + actualBuf.length + ", expect: " + expectBuf.length,
+        actualBuf.length == expectBuf.length);
+    for (int i = 0; i < actualBuf.length; i++) {
+      assertEquals(notEqualPrint(i, actualBuf, expectBuf), expectBuf[i], actualBuf[i]);
+    }
+  }
+
+  private byte[] inputStreamToByteArray(InputStream is) throws IOException {
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+    int nRead;
+    byte[] data = new byte[16384];
+
+    //should be '!= -1' instead of '> 0', but MessageFormatInputStream
+    //currently doesn't return -1 with the read(byte[], int, int) method
+    while ((nRead = is.read(data, 0, data.length)) > 0) {
+      buffer.write(data, 0, nRead);
+    }
+    buffer.flush();
+    return buffer.toByteArray();
+  }
+
+  private void verifyOutput(StoreCopier.Message output, StoreCopier.Message expected) throws IOException {
+    assertTrue("MessageInfos not equal, Output: " + output.getMessageInfo() + " Expected: " + expected.getMessageInfo(),
```

Comment:
use `assertEquals()`? Will print the args too if not equal

Reply:
changed